server/controllers/finance/debtorGroups.js

```diff
@@ -0,0 +1,176 @@
+/**
+* The Debtor Groups Controllers
+*
+* @module finance/debtorGroups
+*
+* @desc This module is responsible for handling all crud operations relatives
+* to debtor groups, and relatives functions
+*
+* @required q
+* @required lib/db
+* @required lib/guid
+*
+* @todo implements the creation of a Debtor Group
+*/
+
+'use strict';
+
+var q  = require('q'),
+    db = require('../../lib/db'),
+    guid = require('../../lib/guid');
+
+/** Get debtor group details */
+exports.detail = getDetail;
+
+/** Get the list of debtor group */
+exports.list = getList;
+
+/** [HTTP API ENDPOINT] Get debtor groups invoices list */
+exports.fetchInvoices = fetchInvoices;
+
+/** [SERVER SIDE FUNCTION] Get debtor groups invoices list */
+exports.getInvoices = getInvoices;
+
+/**
+* GET /debtor_groups/:uuid
+*
+* @function getDetail
+*
+* @desc This function is responsible for retrieving details of a debtor group
+*/
+function getDetail(req, res, next) {
+  var uuid = req.params.uuid;
+
+  var debtorDetailsQuery =
+    'SELECT uuid, name, account_id, location_id, phone, email, note, locked ' +
+      'max_credit, is_convention, price_list_uuid ' +
+    'FROM debitor_group ' +
+    'WHERE uuid = ?';
+
+  db.exec(debtorDetailsQuery, [uuid])
+  .then(function (rows) {
+    if (!rows.length) { return next(new req.codes.ERR_NOT_FOUND()); }
+    res.status(200).json(rows[0]);
+  })
+  .catch(next)
+  .done();
+
+}
+
+/**
+* GET /debtor_groups/
+* GET /debtor_groups/?with_locked={1|0}
+*
+* @function getList
+*
+* @desc This function is responsible for retrieving details of a debtor group
+*/
+function getList(req, res, next) {
+  var query;
+
+  query = 'SELECT uuid, name, locked, account_id FROM debitor_group ';
+  query += (req.query.with_locked === '1') ? '' : 'WHERE locked = 0';
+
+  db.exec(query)
+  .then(function (rows) {
+    res.status(200).json(rows);
+  })
+  .catch(next)
+  .done();
+}
+
+/**
+* GET /debtor_groups/{:uuid}/invoices
+* GET /debtor_groups/{:uuid}/invoices?balanced=1
+*
+* @function fetchInvoices
+*
+* @desc This function is responsible for getting all invoices of a specified debtor group
+*/
+function fetchInvoices(req, res, next) {
```

Comment:
Hmm...  You have two functions `fetchInvoices()` and `getInvoices()`.  In the future, I would put the `getInvoices()` function as the HTTP handler, and the `fetchInvoices()` as the server side function.  That way `GET /debtor_groups/:uuid/invoices` maps to `getInvoices`, which is a very obvious connection.